manual/Makefile

```diff
@@ -1,32 +1,37 @@
-all: tools
-	cd manual; ${MAKE} all
-	${MAKE} tests
-#	cd fpcl; ${MAKE} all
+all: tools manual tests
 
-clean:
-	cd manual; ${MAKE} clean
-	cd tools; ${MAKE} clean
-#	cd fpcl; ${MAKE} clean
 
-release:
-	cd manual; ${MAKE} release
-#	cd fpcl; ${MAKE} release
+tests: manual
+	$(MAKE) -C tests all
 
-.PHONY: tools
 tools:
-	cd tools; ${MAKE} clean; ${MAKE} all
+	$(MAKE) -C tools all
```

Comment:
Why did you decide to drop the `make clean` step here?

Reply:
Yes, I'm not really sure why the tools used to get rebuilt each time because I'm pretty sure they don't depend on anything that changes with each rebuild of the manual.

Reply:
The `make clean` makes sense from the perspective of someone who changes the compiler codebase often, and builds the manual rarely. What's at risk of changing is not the manual build artefacts, but the compiler itself (which is used to build the tools). If the manual/tools Makefile hasn't properly specified its dependencies on the compiler tools (which is hard to do: compilers, standard library, etc.), then it is safer to just rebuild them each time we rebuild the manual, to guarantee that they have been built with the latest compiler version. Otherwise you could get into weird issues when, for example, we change the executable magic numbers, where the tools are compiled with an older compiler version and they refuse to run with the new OCaml runtime during the manual build.

(Then it makes sense to have this clean in this `manual/` Makefile rather than in the `all:` rule in `manual/tools`, as the latter is likely to be used by people who quickly iterate changing the tools, while the former is used by everyone trying to build the manual and not familiar with the subtleties of the build system.)

To summarize: there could be a better way to do this, but it sort of makes sense to me that the `clean` rule is there.

Reply:
Right, I didn't consider that. It shouldn't add much to the build times anyway, since there's not many tools to build anymore.

Reply:
Just an aside: while fixing the manual, I often re-build it many times in an hour, so getting the dependencies right eventually would be best.